docs/messaging/messaging-metrics.md

```diff
@@ -179,6 +180,20 @@ _Note: The need to report `messaging.process.messages` depends on the messaging
 | `messaging.process.messages` | Counter | `{message}` | Measures the number of processed messages. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 <!-- endsemconv -->
 
+### Metric: `messaging.consumer.latency.duration`
+
+This metric is [recommended][MetricRecommended] for any consumer with the capability to extract these timings.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10, 30, 60, 300, 600, 1800 ]`.
+
+<!-- semconv metric.messaging.consumer.latency.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `messaging.consumer.latency.duration` | Histogram | `s` | Measures the duration between message production and consumption. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
> the duration between message production and consumption

I wonder if we'd need to define this more exact. Some messaging systems provide you with the time a message was created ("message creation time" via the client API), others provide you with the time a message was into a partition ("message enqueue time" in the broker).

I'm not sure if it's feasible to specify exact semantics that are applicable across all messaging systems, however, I think it's beneficial to have consistent directions for each specific messaging system (to have consistent semantic and implementations across different client library implementations of the same system).

Reply:
This came up in a few comments.  I tried to address this in my latest changes by declaring two different latencies that represent "creation to processing" time and "enqueued to processing" time. I proposed some specific names: `messaging.latency.duration` and `messaging.buffering.duration` , respective, but I'm open to suggestions.